### PR TITLE
Upgrade polkadot-sdk v`1.3.0 -> 1.6.0`

### LINKED ISSUE #39: 🎁 [Feature Request]: Replace deprecated DmpQueue with MessageQueue
### What is the feature you would like to see?

See most recent version of the cumulus parachain template to see how this should be done

Blocked until next dep upgrade

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### LINKED ISSUE #92: [dependency upgrade] `polkadot-v1.3.0` -> `polkadot-v1.6.0`
### What happened?

1. Upgrade the versions for all polkadot dependencies from `polkadot-v1.3.0` to `polkadot-v1.6.0`. 
2. Summarize important relevant changes from the polkadot releases.

### platform

- [ ] linux
- [ ] windows
- [ ] macos

### Expected behavior

_No response_

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by peektism**
Closes #92 
Closes #39

1. Removes spurious error emission:
```
Nov 21 13:37:34.593 ERROR parachain::pvf: 🚨 Your system cannot securely run a validator. 
Running validation of malicious PVF code has a higher risk of compromising this machine.
  - Cannot unshare user namespace and change root, which are Linux-specific kernel security features: not available: Nov 21 13:37:34.582 TRACE parachain::pvf-common: unsharing the user namespace and calling pivot_root worker_kind=check pivot root worker_pid=1562989 worker_dir_path="/tmp/.tmpxnRzK7/check-can-unshare4LXkykVpKt"
mount MS_BIND: No such file or directory (os error 2)
Nov 21 13:37:34.594 ERROR parachain::pvf: In the next release this will be a hard error by default.
     
More information: https://wiki.polkadot.network/docs/maintain-guides-secure-validator#secure-validator-mode
```
If seen after this upgrade, error log is legit.

2. [XCM builder for XCM construction](https://github.com/paritytech/polkadot-sdk/pull/2107)
3. [pallet-xcm: enhance `reserve_transfer_assets` to support remote reserves](https://github.com/paritytech/polkadot-sdk/pull/1672)
4. [Add `sudo::remove_key`](https://github.com/paritytech/polkadot-sdk/pull/2165)
5. [Improvements to `try-state` devex](https://github.com/paritytech/polkadot-sdk/pull/2019)

1. [Small fixes to beefy consensus](https://github.com/paritytech/polkadot-sdk/pull/2378)
2. [Extend XCM builders](https://github.com/paritytech/polkadot-sdk/pull/2253)

1. [More XCM logging for easier debugging](https://github.com/paritytech/polkadot-sdk/pull/2799)
2. [pallet-assets swap credit](https://github.com/paritytech/polkadot-sdk/pull/1677)
3. [Accept Root origin for ensure_sudo](https://github.com/paritytech/polkadot-sdk/pull/2783)

### CODE REVIEW

File: .github/workflows/cargo-tests.yml [hunk 1]
```diff
@@ -54,6 +54,9 @@ jobs:
       - name: install rust toolchain
         uses: actions-rust-lang/setup-rust-toolchain@v1
 
+      - name: Add required rustup component
+        uses: rustup component add rust-src 
+
```
Comment (ozgunozerk) [inline .github/workflows/cargo-tests.yml:59]: Any explanation on why these are required?

Comment (ozgunozerk) [reply .github/workflows/cargo-tests.yml:59]: Ok found it: https://github.com/OpenZeppelin/polkadot-runtime-template/actions/runs/7454542903/job/20282042449

Probably due to new changes 👍

File: .github/workflows/cargo-tests.yml [hunk 2]
```diff
@@ -54,6 +54,9 @@ jobs:
       - name: install rust toolchain
         uses: actions-rust-lang/setup-rust-toolchain@v1
 
+      - name: Add required rustup component
+        uses: rustup component add rust-src
```
Comment (ozgunozerk) [inline .github/workflows/cargo-tests.yml:58]: This breaks the CI.

> the `uses' attribute must be a path, a Docker image, or owner/repo@ref

**Review by KitHat — APPROVED** [review]

File: node/src/service.rs [hunk 3]
```diff
@@ -257,10 +257,15 @@ async fn start_node_impl(
...
-            );
+        match SUBSTRATE_REFERENCE_HARDWARE.check_hardware(&hwbench) {
+            Err(err) if validator => {
+                log::warn!(
+                    "⚠️  The hardware does not meet the minimal requirements {} for role \
+                     'Authority'.",
+                    err
+                );
+            }
+            _ => {}
```
Comment (ozgunozerk) [inline node/src/service.rs:268]:
```suggestion
        if let Err(err) = SUBSTRATE_REFERENCE_HARDWARE.check_hardware(&hwbench) {
            if validator {
                log::warn!(
                    "⚠️  The hardware does not meet the minimal requirements {} for role \
                     'Authority'.",
                    err
                );
            }
        }
```

Having an empty match arm feels like we are using the wrong tool. I think `if let(err)` suits better for this. Or if we do not need to extract the error, even a simple `if` with `&&` to combine two if statements.

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -128,89 +123,72 @@ pub fn local_testnet_config() -> ChainSpec {
...
-        // Telemetry
-        None,
-        // Protocol ID
-        Some("template-local"),
-        // Fork ID
-        None,
-        // Properties
-        Some(properties),
-        // Extensions
+    #[allow(deprecated)]
```
Comment (ozgunozerk) [inline node/src/chain_spec.rs:126]: We are doing an upgrade, the code should be newer, it surprised me to see introduce deprecated code in an upgrade PR. 
Was this necessary for the compatibility with the new versions?

**Review by ozgunozerk — APPROVED** [review]
Superb documentation, thanks, will be very useful in the future to see what has changed between versions.

I think we can merge this as soon as open conversations are closed 👍

File: node/src/service.rs (hunk 3 above)
Comment (ozgunozerk) [reply node/src/service.rs:268]: Hmm, i'm surprised we are on different sides about this one. Doesn't matter though, very minor thing. Not a blocker :)

Out of curiosity, if you can share the example from polkadot-sdk for the match (where one arm is empty and can be replaced with `if let Err()`, I'd be glad.

**Review by KitHat — APPROVED** [review]

**Pull request merged**